packages/grid/_modules_/grid/utils/material-ui-utils.ts

```diff
@@ -105,7 +105,7 @@ const globalPseudoClassesMapping: Record<string, string> = {
 
 export function generateUtilityClass(componentName: string, slot: string): string {
   const globalPseudoClass = globalPseudoClassesMapping[slot];
-  return globalPseudoClass || `${componentName}-${slot}`;
+  return globalPseudoClass || `${componentName}-${slot}`.replace('__', '--');
```

Comment:
I got a bit confused about the usage of `__` for the JS classes API, and `--` for the CSS class name API. Is the long-term plan to unify around `__`? I recall seeing an issue but I can't find it back. Maybe to open a new one.

Reply:
I reverted this part because I didn't receive many positive feedbacks about it in https://github.com/mui-org/material-ui-x/issues/420#issuecomment-896359219. We could re-evaluate if the BEM syntax is something we want to use in the future. I was trying to avoid having to use this syntax: https://github.com/mui-org/material-ui-x/pull/2458/files#diff-daa7ce589b380e02bf73ba2d9b08377cc7571039f62c5e057e027f3f0393779aR86